### ISSUE TITLE
QR is not stable, Unable to connect QR

### ISSUE THREAD

**Issue by RAKESHSUVIDYA** [pedroslopez/whatsapp-web.js#3021]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

I am using this following code:
client= new Client({
              authStrategy: new LocalAuth({
                clientId: id,
                dataPath: "./sessions",
              }),
              webVersionCache: {
                type: "remote",
                remotePath:
                  "https://raw.githubusercontent.com/wppconnect-team/wa-version/main/html/2.2412.54.html",
              },
              puppeteer: {
                headless: true,
                args: [
                  "--no-sandbox",
                  "--no-first-run",
                  "--disable-setuid-sandbox",
                  "--disable-dev-shm-usage",
                  "--disable-accelerated-2d-canvas",
                  "--disable-gpu",
                  "--single-process",
                  "--no-zygote",
                ],
              },
            });

It was working yesterday but now it is not working. I mean to say that qr connection error.
![WhatsApp Image 2024-05-11 at 10 32 27_f31a17c3](https://github.com/pedroslopez/whatsapp-web.js/assets/16055762/43eeba5d-0856-4fc5-92c9-4bcb055b0c08)

### Expected behavior

QR Code is changing frequently. It is not stable. Unable to connect QR

### Steps to Reproduce the Bug or Issue

I have tried followed https://github.com/pedroslopez/whatsapp-web.js/pull/2816. but not able to solve

### Browser Type

Chromium

### WhatsApp Account Type

Standard

### Does your WhatsApp account have multidevice enabled?

Yes, I am using Multi Device

### Environment

OS: Ubuntu
Phone: Android
whatsapp-web.js version 1.23.0
WhatsApp Web version [run await client.getWWebVersion()]:
Node.js Version v20.12.2

**Comment by patharitech** [comment]
I can see this as marked completed, can you please let me know the root cause or how to fix it. I can see the same happening for my ubuntu instance but works fine in windows env.

**Comment by RAKESHSUVIDYA** [comment]
It is not still resolved. I do not know why he has closed this issues.

**Comment by sanskari27** [comment]
QRCode changes just after being scanned and new qr string starts with undefined.
Although it is working fine in mac and windows but it doesn't work un ubuntu environment.

Please help out with a reason or a solution if possible..

**Comment by bhejosms** [comment]
Facing same issue

Qr changes as below

QR Code IS : 2@JnJbLx6xjPhN3QiY+7ggYg/jI4BJ61CTqTbn9KgSp4y5OSjdSEfr4anjBLORSXKhe1LpXX1YJjtaEQ==,10eqtGg5dxIInWS+psdT12Rb6BEA4sjm/BIJ+6F4M1c=,0ju3FrR9fPRcosvhUaTJllY8CSi44yBJFvtiJnY5zWY=,Bl0S9ssWVRs4MJ3wErjtcWZ4+Z5aDi5AjUBE/aw+uNk=,1
/qr
QR Code IS : undefined,10eqtGg5dxIInWS+psdT12Rb6BEA4sjm/BIJ+6F4M1c=,0ju3FrR9fPRcosvhUaTJllY8CSi44yBJFvtiJnY5zWY=,Bl0S9ssWVRs4MJ3wErjtcWZ4+Z5aDi5AjUBE/aw+uNk=,1
/qr
/qr
/qr
/qr
/qr
/qr
/qr
/qr

**Comment by RAKESHSUVIDYA** [comment]
this is the problem in ubuntu os

**Comment by gaurav23122001** [comment]
Hi @RAKESHSUVIDYA, how did you run your server then. Which OS did you shift to?

**Comment by RAKESHSUVIDYA** [comment]
> Hi @RAKESHSUVIDYA, how did you run your server then. Which OS did you shift to?

It is working fine in windows os. I have check locally in my system.

**Comment by andrscyv** [comment]
I'm getting this error

**Comment by gaurav23122001** [comment]
@RAKESHSUVIDYA Have you tried on windows server or were able to run in any other linux vps?

**Comment by RAKESHSUVIDYA** [comment]
> @RAKESHSUVIDYA Have you tried on windows server or were able to run in any other linux vps?

Yes I have check in other vps linux. 
Issues are continue.

**Comment by janke184** [comment]
same issue here!

**Comment by fathah** [comment]
Any update on this issue

**Comment by gaurav23122001** [comment]
It works on windows machine locally but I tested on windows server 2022
It doesn't work on that too

**Comment by fathah** [comment]
At the time I didn't get much time to wait for the fix or look for bugs, I ran the authentication in Local Windows Machine (Win11). Once authentication is done, I committed the LocalAuth cache folder to my server. 
*Not a solution, just quick way.*

### .gitignore
```

/.wwebjs_auth/session/Default/Cache/
/.wwebjs_auth/session/Default/chrome_debug.log
/.wwebjs_auth/session/Default/File System/Origins/CURRENT
/.wwebjs_auth/session/Default/File System/Origins/LOG
/.wwebjs_auth/session/Default/File System/Origins/LOG.old
/.wwebjs_auth/session/Default/File System/000/t/Paths/LOG
/.wwebjs_auth/session/Default/File System/000/t/Paths/CURRENT
/.wwebjs_auth/session/Default/GPUCache/
```

**Comment by andrscyv** [comment]
Why is this closed ?

**Comment by Hasnen-mr** [comment]
is this issue resolved?
why is this closed 
still in Ubuntu Indian numbers are not able to scan.
can you help which windows version is required to setup Puppiter in windows?

**Comment by gaurangDhameliya** [comment]
Hello, Guy
Is there any solution for this issue?

**Comment by janke184** [comment]
Try to connect using a VPN in your phone:

You can use those applications:

https://apps.apple.com/in/app/vpn-usa-for-master-uk-proxy/id1460571281

https://play.google.com/store/apps/details?id=free.vpngermany.hotspot.unlimited.proxy.turbo.secure

Connect using a NY location.

Regards.

**Comment by tobizinho** [comment]
Pessoal, isso foi resolvido?

O erro está acontecendo no meu Linux POP OS

**Comment by fathah** [comment]
Any update on this issue?

**Comment by prashant1879** [comment]
I face same issue. 
on local machine ( Ubuntu 20.x its works fine ) while upload on server its not works .
nodejs version same as local
os version same as local 

my VPS server location is germany and after connect VPN of germany (https://play.google.com/store/apps/details?id=free.vpngermany.hotspot.unlimited.proxy.turbo.secure&pli=1) dont get that undefinded issue.

**Comment by prashant1879** [comment]
Hello everyone & @RAKESHSUVIDYA ,

I recently faced the same issue with my setup, which has now been resolved. Here’s what happened:

Despite having the same packages and node version on Ubuntu 20.x, my setup didn't work when the VPS was located in Germany. However, after switching to a VPS located in India, everything works perfectly fine without any further changes.
It appears that the problem might be related to the remote VPS location. For those based in India, using a VPS located in 
Germany (or other distant locations) might cause similar issues. If you encounter such problems, I recommend trying a VPS closer to your physical location.

I'm available for help if anyone requires just ping me on [skype](https://join.skype.com/invite/ji4GdXi1ti8x)

![image_2024_06_03T12_20_00_718Z (1)](https://github.com/pedroslopez/whatsapp-web.js/assets/19793864/33147bed-74af-464a-9231-1d09aced403e)
